Rubberduck.Inspections/Concrete/UnreachableCaseInspection/UCIRangeClauseFilter.cs

```diff
@@ -0,0 +1,1000 @@
+                || filter.RangeValues.Count != RangeValues.Count
+                || filter.VariableRanges.Count != VariableRanges.Count
+                || filter.RelationalOps.Count != RelationalOps.Count)
+            {
+                return false;
+            }
+
+            var clausesMatch = true;
+            if (filter.TryGetIsLTValue(out T isLT))
+            {
+                if (TryGetIsLTValue(out T myLT))
+                {
+                    if (isLT.CompareTo(myLT) != 0)
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            if (filter.TryGetIsGTValue(out T testGT))
+            {
+                if (TryGetIsGTValue(out T myGT))
+                {
+                    if (testGT.CompareTo(myGT) != 0)
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            if (!clausesMatch)
+            {
+                return false;
+            }
+            var theRanges = filter._ranges.All(rg => _ranges.Contains(rg))
+                    && filter._variableRanges.All(vrg => _variableRanges.Contains(vrg));
+
+            var singles = filter._singleValues.All(rg => _singleValues.Contains(rg));
+            var relOps = filter._relationalOps.All(ro => _relationalOps.Contains(ro));
+            return theRanges && relOps && singles;
+        }
+
+        public override string ToString()
+        {
+            var descriptors = new List<string>();
+            descriptors = AddDescriptor(GetIsClausesDescriptor(CompareTokens.LT), descriptors);
+            descriptors = AddDescriptor(GetIsClausesDescriptor(CompareTokens.GT), descriptors);
+            descriptors = AddDescriptor(GetRangesDescriptor(), descriptors);
+            descriptors = AddDescriptor(GetSinglesDescriptor(), descriptors);
+            descriptors = AddDescriptor(GetRelOpDescriptor(), descriptors);
+            var descriptor = string.Empty;
+            foreach (var desc in descriptors)
+            {
+                descriptor = descriptor + desc + "!";
+            }
+            if (descriptor.Length > 0)
+            {
+                return descriptor.Substring(0, descriptor.Length - 1);
+            }
+            return string.Empty;
+        }
+
+        private static List<string> AddDescriptor(string descriptor, List<string> content)
+        {
+            if (descriptor.Length > 0)
+            {
+                content.Add(descriptor);
+            }
+            return content;
+        }
+
+        public override int GetHashCode()
+        {
+            return ToString().GetHashCode();
+        }
+
+        internal void AddExtents(IUCIValue min, IUCIValue max)
```

Comment:
`protected` or `private` does not compile.  This method is called by the class' factory...I made it `internal` because `public` seemed too accessible for a method I only wanted called by the factory.  That said, there is little reason not to make it `public`, but keep it off the primary interface.